pkg/controller/volume/attachdetach/cache/actual_state_of_world.go

```diff
@@ -677,10 +671,48 @@ func (asw *actualStateOfWorld) GetVolumesToReportAttached() map[types.NodeName][
 	return volumesToReportAttached
 }
 
+func (asw *actualStateOfWorld) GetVolumesToReportAttachedForNode(nodeName types.NodeName) (bool, []v1.AttachedVolume) {
```

Comment:
it seems to me the returned bool is not very necessary. Based on reconciler logic, the only possible case is the node is no longer in actual state. In this case, volume list will be empty. I am wondering in this case, we should update node status to have empty volume list or not. Or Maybe this case of node empty is also should not exist?

Reply:
If I don't report this bool, then I have no way of knowing if you need to update the node status.volumesAttached property with an empty list or if there is nothing to do (for instance when we detach the last volume from the node)

Reply:
What I mean is GetVolumesToReportAttachedForNode() is only needed by UpdateNodeStatusForNode(), which is only called in reconciler during detach, and before calling it, it mark volume/node by RemoveVolumeFromReportAsAttached. In this case, the bool will be always true, right? If this function is used in other places, we need to have bool value, I think.
 https://github.com/kubernetes/kubernetes/pull/108167/files#diff-e76cff1d139f1df6d8c37cea61b19aa8caea6b103d64e58a084f2d81c5bad149R196